diff --git a/pom.xml b/pom.xml
index ce92ea5..07a81c8 100644
--- a/pom.xml
+++ b/pom.xml
@@ -220,17 +220,19 @@
           </execution>
         </executions>
       </plugin>
+<!--
       <plugin>
         <groupId>org.graalvm.buildtools</groupId>
         <artifactId>native-maven-plugin</artifactId>
       </plugin>
+-->
       <plugin>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-maven-plugin</artifactId>
         <executions>
           <execution>
             <!-- Spring Boot Actuator displays build-related information
-              if a META-INF/build-info.properties file is present -->
+            if a META-INF/build-info.properties file is present -->
             <goals>
               <goal>build-info</goal>
             </goals>
@@ -241,8 +243,24 @@
                 <java.source>${java.version}</java.source>
                 <java.target>${java.version}</java.target>
               </additionalProperties>
+
+              <image>
+                <builder>paketobuildpacks/builder:tiny</builder>
+                <env>
+                  <BP_NATIVE_IMAGE>true</BP_NATIVE_IMAGE>
+                </env>
+              </image>
+
             </configuration>
           </execution>
+
+          <execution>
+            <id>process-aot</id>
+            <goals>
+              <goal>process-aot</goal>
+            </goals>
+          </execution>
+
         </executions>
       </plugin>
       <plugin>
diff --git a/src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java b/src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java
index ac6e150..3ef7b5c 100644
--- a/src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java
+++ b/src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java
@@ -30,8 +30,67 @@ import org.springframework.context.annotation.ImportRuntimeHints;
 @ImportRuntimeHints(PetClinicRuntimeHints.class)
 public class PetClinicApplication {
 
+	static void iteration(int i, String[] args) {
+		long begin = System.currentTimeMillis();
+		org.springframework.context.ConfigurableApplicationContext ctx = SpringApplication
+			.run(PetClinicApplication.class, args);
+		long end = System.currentTimeMillis();
+		System.out.println(i + ": #### Booted and returned in " + (end - begin) + "ms");
+		ctx.close();
+	}
+
+	static void iteration0(String[] args) {
+		iteration(0, args);
+	}
+
+	static void iteration1(String[] args) {
+		iteration(1, args);
+	}
+
+	static void iteration2(String[] args) {
+		iteration(2, args);
+	}
+
+	static void iteration3(String[] args) {
+		iteration(3, args);
+	}
+
+	static void iteration4(String[] args) {
+		iteration(4, args);
+	}
+
+	static void iteration5(String[] args) {
+		iteration(5, args);
+	}
+
 	public static void main(String[] args) {
-		SpringApplication.run(PetClinicApplication.class, args);
+		Integer maxIters = Integer.getInteger("iterations", 1);
+		for (int i = 0; i < maxIters; i++) {
+			switch (i) {
+				case 0:
+					iteration0(args);
+					break;
+				case 1:
+					iteration1(args);
+					break;
+				case 2:
+					iteration2(args);
+					break;
+				case 3:
+					iteration3(args);
+					break;
+				case 4:
+					iteration4(args);
+					break;
+				case 5:
+					iteration5(args);
+					break;
+
+				default:
+					iteration(i, args);
+					break;
+			}
+		}
 	}
 
 }
